### PR TITLE
Pre commit Hook added.  

### LINKED ISSUE #196: Automate tests in pre-commit hook script.
Currently we dont have any scripts to check for linting/ formatting error in local. I propose we should add a script to extract certain steps/command directly from .github/workflow  and execute them before committing. This will eliminate extra commit fixing the lint and format. This will also ensure each commit is independent and does not give any error. We can further expand it to run the tests before any push to ensure each commit is complete without any errors.

what we need:
A script to extract steps/commads ( probably python script  )

A script to execute the commands ( bash script ) `.precommit script` which can be symlink with git/hook/precommit 
`ln -s ../../.pre-commit .git/hooks/pre-commit` . This will automatically trigger on commit.

### PR DESCRIPTION
**Pull request by aryan-bhokare**
closes #196 

### Changes Introduced:
- Automated testing and pre-commit hook integration to ensure code quality and adherence to linting standards.
- Added a GitHub Actions workflow file (`build.yml`) to facilitate automated testing and pre-commit hook execution.
- Created a custom script (`.github-workflow-script`) to extract specific steps from the workflow file that contain the `COMMIT_HOOKS` environment variable set to "pre-commit".

### Details:
- The GitHub Actions workflow defined in `build.yml` is triggered upon pull requests targeting the "master" branch.
- The workflow includes a job named "build" that sets up the Go environment, performs Go formatting checks, runs linting checks using golangci-lint, and builds the project.
- The `gofmt` check step is conditioned by the presence of the `COMMIT_HOOKS` environment variable set to "pre-commit".
- Upon triggering the workflow, the steps containing the `COMMIT_HOOKS` variable are extracted and executed as part of the pre-commit hook integration process.
- This setup aims to streamline the development process by automating testing and enforcing code quality standards at each commit, contributing to a cleaner and more maintainable codebase.

### Prerequisite:
use `ln -s ../../.pre-commit .git/hooks/pre-commit` to create a symbolic link named `pre-commit` in the `.git/hooks` directory that points to the `.pre-commit` .

### CODE REVIEW

**Review by kartikaysaxena — COMMENTED** [review]
I think maybe the hooks should be in the .git/hooks dir, but again, it seems that git doesn't track that as it is created automatically when you fork a repo. Also, I see the scripts are executed only when the `build.yml` file is triggered, which means only when someone pushes it to the repository which shouldn't be the case.

I tried it in your code in a  codespace but it is still letting me commit some empty lines.

**Comment by shivam-Purohit** [comment]
@aryan-bhokare  have you tested the code in the local? if you have would you share ss or some example (otherwise i have to check it myself) Other than that i think we should follow this approach and add more checks. @SarthakJain26 @neelanjan00 what more checks we should add before committing other than linting and fmt?

**Comment by shivam-Purohit** [comment]
> I tried it in your code in a codespace but it is still letting me commit some empty lines.

can you try and tell after linking it with pre-commit hook in git like this?
`ln -s ../../.pre-commit .git/hooks/pre-commit`

i think this should work if you have already forked it, can you check it if its working?

**Comment by kartikaysaxena** [comment]
> > I tried it in your code in a codespace but it is still letting me commit some empty lines.
> 
> 
> 
> can you try and tell after linking it with pre-commit hook in git like this?
> 
> `ln -s ../../.pre-commit .git/hooks/pre-commit`
> 
> 
> 
> i think this should work if you have already forked it, can you check it if its working?

Yeah I didn't try this step as it wasn't listed as a prerequisite in the description

**Comment by aryan-bhokare** [comment]
Making Formatting Error 

Screenshot from 2024-02-22 10-21-44

Result
Screenshot from 2024-02-22 10-20-51

File: pkg/cmd/get/environment.go [hunk 1]
```diff
@@ -52,7 +52,7 @@ var ChaosEnvironmentCmd = &cobra.Command{
 
 		environmentID, err := cmd.Flags().GetString("environment-id")
 		utils.PrintError(err)
-
+		utils.Red.Println("i am making something to check")
```
Comment (SarthakJain26) [inline pkg/cmd/get/environment.go:55]: This can be removed if you are done making something to check 😛

**Review by Saranya-jena — APPROVED** [review]

**Review by SarthakJain26 — APPROVED** [review]

**Review by Jonsy13 — APPROVED** [review]

**Pull request merged**